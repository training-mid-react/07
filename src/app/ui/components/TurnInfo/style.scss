@use '../../styles/mixins' as mix;

.turn-info {
    align-items: center;
    gap: 10px;
    justify-content: center;
    @include mix.flex(row);
  }