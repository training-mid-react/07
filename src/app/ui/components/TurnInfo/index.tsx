import React from 'react';
import Title from '@components/Title';
import './style.scss';

interface TurnInfoProps {
  currentPlayer: string;
}

const TurnInfo: React.FC<TurnInfoProps> = ({ currentPlayer }) => {
  return (
    <div className="turn-info" data-testid="turn-info">
      <Title as="h3">
        Turno de {currentPlayer}
      </Title>
    </div>
  );
};

export default TurnInfo;
