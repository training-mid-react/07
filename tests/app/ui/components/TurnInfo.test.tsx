import React from 'react';
import { render } from '@testing-library/react';
import TurnInfo from '../../../../src/app/ui/components/TurnInfo';

describe('Componente TurnInfo', () => {
  const currentPlayer = 'Player 1';

  it('renderiza correctamente', () => {
    const { asFragment } = render(<TurnInfo currentPlayer={currentPlayer} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renderiza el contenedor principal con el data-testid correcto', () => {
    const { getByTestId } = render(<TurnInfo currentPlayer={currentPlayer} />);
    const turnInfoElement = getByTestId('turn-info');
    expect(turnInfoElement).toBeInTheDocument();
  });

  it('renderiza el tÃ­tulo dentro de un h3', () => {
    const { container } = render(<TurnInfo currentPlayer={currentPlayer} />);
    const h3Element = container.querySelector('h3');
    expect(h3Element).toBeInTheDocument();
    expect(h3Element).toHaveTextContent(`Turno de ${currentPlayer}`);
  });

  it('renderiza el texto del turno correctamente', () => {
    const { getByText } = render(<TurnInfo currentPlayer={currentPlayer} />);
    const textElement = getByText(`Turno de ${currentPlayer}`);
    expect(textElement).toBeInTheDocument();
  });
});
