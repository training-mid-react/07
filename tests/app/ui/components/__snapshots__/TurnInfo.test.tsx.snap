// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente TurnInfo > renderiza correctamente 1`] = `
<DocumentFragment>
  <div
    class="turn-info"
    data-testid="turn-info"
  >
    <h3
      class="title"
      data-testid="title-component"
    >
      Turno de Player 1
    </h3>
  </div>
</DocumentFragment>
`;
